opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1032,150 +1033,98 @@ namespace Opm {
+            if (active_[ Oil ] && active_[ Gas ]) {
                 for (int c = 0; c < nc; ++c) {
                     const int region = fipnum[c] - 1;
                     if (region != -1) {
-                        const auto& intQuants = *ebosSimulator_.model().cachedIntensiveQuantities(c, /*timeIdx=*/0);
-                        const auto& fs = intQuants.fluidState();
-                        const double hydrocarbon = fs.saturation(FluidSystem::oilPhaseIdx).value() + fs.saturation(FluidSystem::gasPhaseIdx).value();
-                        hcpv[region] += pv[c] * hydrocarbon;
-                        pres[region] += pv[c] * fs.pressure(FluidSystem::oilPhaseIdx).value();
-                    }
-                }
-                for (int c = 0; c < nc; ++c) {
-                    const int region = fipnum[c] - 1;
-                    if (region != -1) {
-
-                        fip_.fip[FIPData::FIP_PV][c] = pv[c];
-                        const auto& intQuants = *ebosSimulator_.model().cachedIntensiveQuantities(c, /*timeIdx=*/0);
-                        const auto& fs = intQuants.fluidState();
-                        const double hydrocarbon = fs.saturation(FluidSystem::oilPhaseIdx).value() + fs.saturation(FluidSystem::gasPhaseIdx).value();
-
-                        //Compute hydrocarbon pore volume weighted average pressure.
-                        //If we have no hydrocarbon in region, use pore volume weighted average pressure instead
-                        if (hcpv[region] != 0) {
-                            fip_.fip[FIPData::FIP_WEIGHTED_PRESSURE][c] = pv[c] * fs.pressure(FluidSystem::oilPhaseIdx).value() * hydrocarbon / hcpv[region];
-                        } else {
-                            fip_.fip[FIPData::FIP_WEIGHTED_PRESSURE][c] = pres[region] / pv[c];
-                        }
-
-                        values[region][FIPData::FIP_PV] += fip_.fip[FIPData::FIP_PV][c];
-                        values[region][FIPData::FIP_WEIGHTED_PRESSURE] += fip_.fip[FIPData::FIP_WEIGHTED_PRESSURE][c];
+                        values[region][FIPData::FIP_DISSOLVED_GAS] += fip_.fip[FIPData::FIP_DISSOLVED_GAS][c];
+                        values[region][FIPData::FIP_VAPORIZED_OIL] += fip_.fip[FIPData::FIP_VAPORIZED_OIL][c];
                     }
                 }
             }
-            else
-            {
-#if HAVE_MPI
-                // mask[c] is 1 if we need to compute something in parallel
-                const auto & pinfo =
-                        boost::any_cast<const ParallelISTLInformation&>(istlSolver().parallelInformation());
-                const auto& mask = pinfo.updateOwnerMask( fipnum );
 
-                auto comm = pinfo.communicator();
-                // Compute the global dims value and resize values accordingly.
-                dims = comm.max(dims);
-                values.resize(dims, std::vector<double>(FIPData::fipValues,0.0));
+            comm.sum(hcpv.data(), hcpv.size());
```

Comment:
Seems like these values have not even been initialized yet due to the changed order of the computations. That happens down in line 1088 ff.